bin/check_samplesheet.py

```diff
@@ -1,29 +1,203 @@
 #!/usr/bin/env python
 
 
+import argparse
+import logging
 import os
 import re
 import sys
 import errno
-import argparse
 import re
+from pathlib import Path
+
+
+logger = logging.getLogger()
+
+
+class RowChecker:
+    """
+    Define a service that can validate and transform each given row.
+
+    Attributes:
+        modified (list): A list of dicts, where each dict corresponds to a previously
+            validated and transformed row. The order of rows is maintained.
+
+    """
+
+    VALID_FORMATS = (
```

Comment:
Has this been added by the nf-core template?

We don't use fasts etc, so why is it in there?

Reply:
Also the following code is only suitable for fastq 🤔

Reply:
This is coming in from the nf-core template. Currently we do not use the `RowChecker` class. We could extend and use it. But it does not have to be in this release necessarily in my opinion.

Reply:
To be honest, I don't like code that is not being used :)